iceoryx_posh/include/iceoryx_posh/internal/popo/used_chunk_list.hpp

```diff
@@ -21,161 +22,107 @@
+///        container could be corrupt when the application dies in the wrong moment.
+///        To be able to do the cleanup, RouDi needs to be able to access the list with the used chunk under all
+///        circumstances. This is achieved by storing the ChunkManagement pointer in an array which can always be
+///        accessed. Additionally, the type stored is this array must be less or equal to 64 bit in order to write it
+///        within one clock cycle to prevent torn writes, which would corrupt the list and could potentially crash
+///        RouDi.
+template <uint32_t Capacity>
 class UsedChunkList
 {
-  public:
-    UsedChunkList()
-    {
-        init();
-    }
-
-    // only from runtime context
-    bool insert(mepoo::SharedChunk f_chunk)
-    {
-        if (freeSpaceInList())
-        {
-            // get next free entry after freelistHead
-            auto nextFree = m_list[m_freeListHead];
+    static_assert(Capacity > 0, "UsedChunkList Capacity must me larger than 0!");
 
-            // freeListHead is getting new usedListHead, next of this entry is updated to next in usedList
-            m_list[m_freeListHead] = m_usedListHead;
-            m_usedListHead = m_freeListHead;
+  public:
+    /// @brief Constructs a default UsedChunkList
+    UsedChunkList() noexcept;
+
+    /// @brief Inserts a SharedChunk into the list
+    /// @param[in] chunk to store in the list
+    /// @return true if successful, otherwise false if e.g. the list is already full
+    /// @note only from runtime context
+    bool insert(mepoo::SharedChunk chunk) noexcept;
+
+    /// @brief Removes a chunk from the list
+    /// @param[in] chunkHeader to look for a corresponding SharedChunk
+    /// @param[out] chunk which is removed
+    /// @return true if successfully removed, otherwise false if e.g. the chunkHeader was not found in the list
+    /// @note only from runtime context
+    bool remove(const mepoo::ChunkHeader* chunkHeader, mepoo::SharedChunk& chunk) noexcept;
+
+    /// @brief Cleans up all the remaining chunks from the list.
+    /// @note from RouDi context once the applications walked the plank
+    void cleanup() noexcept;
 
-            // store chunk mgmt ptr
-            m_data[m_usedListHead] = f_chunk.release();
+  private:
+    void init() noexcept;
 
-            // set freeListHead to the next free entry
-            m_freeListHead = nextFree;
+    bool freeSpaceInList() const noexcept;
 
-            //@todo can we do this cheaper with a global fence in cleanup?
-            m_synchronizer.clear(std::memory_order_release);
-            return true;
-        }
-        else
-        {
-            return false;
-        }
-    }
+  private:
+    static constexpr uint32_t INVALID_INDEX{Capacity};
 
-    // only from runtime context
-    bool remove(const mepoo::ChunkHeader* f_chunkHeader, mepoo::SharedChunk& f_chunk)
+    // this shall be moved to the relative_ptr implementation
+    struct relativ_ptr_data
```

Comment:
this should be consolidated with the relative pointer refactoring

Reply:
The renaming too?

Reply:
Everything that is necessary for a lean design with names fitting the general codestyle (I hope we can adopt snake_case for utils classes, in line with STL).